x-pack/plugins/uptime/public/hooks/use_filter_update.ts

```diff
@@ -8,46 +8,70 @@
 import { useEffect } from 'react';
 import { useUrlParams } from './use_url_params';
 
+export const parseFiltersMap = (currentFilters: string): Map<string, string[]> => {
```

Comment:
Can we add some of the comments from the previous file here too, for example
```
// this try catch is necessary to evaluate user input in kuery bar,
```
or
```
 // We are using try/catch here because this is user entered value
  // and JSON.parse and stringifyKueries can have hard time parsing
  // all possible scenarios, we can safely ignore if we can't parse them
```